src/build-data/botan-config.cmake.in

```diff
@@ -0,0 +1,42 @@
+set(_Botan_mod_list "%{mod_list}")
+string(REGEX MATCHALL "[A-Za-z_0-9]+" _Botan_supported_components ${_Botan_mod_list})
+
+foreach(_comp ${Botan_FIND_COMPONENTS})
+  if (NOT _comp IN_LIST _Botan_supported_components)
+    set(Botan_FOUND False)
+    set(Botan_NOT_FOUND_MESSAGE "Unsupported module: ${_comp}")
+  endif()
+endforeach()
+
+%{unless build_shared_lib}
+
+set(Botan_FOUND False)
+set(Botan_NOT_FOUND_MESSAGE "Static build currently not supported with CMake")
+
+%{endif}
+
+%{if build_shared_lib}
+
+if(NOT DEFINED Botan_FOUND)
+    set(Botan_FOUND True)
+endif()
```

Comment:
Ah, I see the logic behind it now. The better way is to:

- unset `Botan_FOUND`
- if a required component is missing, set `Botan_FOUND` to `False` and append a reason to `Botan_NOT_FOUND_MESSAGE`
- after checking components, if `Botan_FOUND` is set, just `return ()`
- this can then be an unconditional `set` to `True`